src/api/process.c

```diff
@@ -133,7 +133,9 @@ static int process_start(lua_State* L) {
     lua_rawget(L, 1);
     cmd[i-1] = luaL_checkstring(L, -1);
   }
-  cmd[cmd_len] = NULL;
+  if (!cmd[0]) {
+    return luaL_error(L, "argv[0] cannot be null");
+  }
```

Comment:
`luaL_checkstring` returns a pointer and thus can return NULL.
If NULL is returned for the first element we can already give up because later on its used to execvp.

Error text could be improved

Reply:
I'm pretty sure luaL_checkstring can't return NULL.

Reply:
you are right, it *shouldn't* return NULL.
But if for whatever reason it does, we are kinda screwed.

It would probably be best to throw an assert in the code instead

Reply:
My point is that, I really don't think it's possible for it to do so, under any circumstances. It's basically checking for a value on the lua stack; which must be a string. If it's a string, I think this always returns a valid pointer. If not, it'll throw a lua error. There's no possibility of it's returning NULL AFAIK.